tests/components/smtp/test_config_flow.py

```diff
@@ -0,0 +1,246 @@
+"""Test smtp config flow."""
+from smtplib import SMTPAuthenticationError
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.smtp.const import DOMAIN
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import MOCK_CONFIG, MOCK_USER_INPUT
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(autouse=True)
+def smtp_setup_fixture():
+    """Patch smtp setup entry."""
+    with patch("homeassistant.components.smtp.async_setup_entry", return_value=True):
+        yield
```

Comment:
I suggest that you move the fixtures to `conftest.py`, and then use `pytestmark` to auto-use them here.
```suggestion
pytestmark = pytest.mark.usefixtures("mock_setup_entry", "mock_client")
```